### ISSUE TITLE
Binder.TryBindIndexOrRangeIndexer doesn't report use-site diagnostics

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#40554]
The method reports some diagnostics and return a bound node via an out parameter.  It declares the following variable:
```
HashSet<DiagnosticInfo> useSiteDiagnostics = null;
```

passes it around to accumulate the diagnostics, but never reports it. The diagnostics should be reported because its presence/absence isn't equivalent to success/failure of the method.
